src/lib/mongoBackend/mongoQueryTypes.cpp

```diff
@@ -236,14 +236,38 @@ HttpStatusCode mongoEntityTypes
    *
    * However, considering that the number of types will be small compared with the number of entities,
    * the current approach seems to be ok
+   *
+   * emptyEntityType is special: it must aggregate results for entity type "" and for entities without type.
+   * Is pre-created before starting processing results and destroyed if at the end it has not been used
+   * (i.e. pushed back into the vector)
+   *
    */
+
+  EntityType* emptyEntityType     = new EntityType("");
+  bool        emptyEntityTypeUsed = false;
+
   for (unsigned int ix = offset; ix < MIN(resultsArray.size(), offset + limit); ++ix)
   {
-    BSONObj                   resultItem  = resultsArray[ix].embeddedObject();
-    EntityType*               entityType  = new EntityType(getStringField(resultItem, "_id"));
-    std::vector<BSONElement>  attrsArray  = getField(resultItem, "attrs").Array();
+    BSONObj                   resultItem  = resultsArray[ix].embeddedObject();    
+    std::vector<BSONElement>  attrsArray  = getFieldF(resultItem, "attrs").Array();
+
+
+    EntityType*               entityType;
 
-    entityType->count = countEntities(tenant, servicePathV, entityType->type);
+    bool nullId = ((resultItem.hasField("")) && (getFieldF(resultItem, "").isNull())) || (getStringFieldF(resultItem, "_id") == "");
+
+    if (nullId)
+    {
+      entityType           = emptyEntityType;
+      emptyEntityTypeUsed  = true;
+    }
+    else
+    {
+      entityType = new EntityType(getStringFieldF(resultItem, "_id"));
+    }
+
+    /* Note we use += due to emptyEntityType accumulates */
+    entityType->count += countEntities(tenant, servicePathV, entityType->type);
```

Comment:
After a second look I think is safe to go without explicit initalization. Looking to the constructors, 0 is set for count:

```
EntityType::EntityType(): count(0)
{

}



/* ****************************************************************************
*
* EntityType::EntityType -
*/
EntityType::EntityType(std::string _type): type(_type), count(0)
{

}
```